core/domain/prod_validation_jobs_one_off.py

```diff
@@ -75,58 +66,153 @@ def _validate_external_id_relationships(cls, item):
                             item.id, field_name, model_id,
                             str(model_class.__name__), model_id))
 
+    @classmethod
+    def _fetch_external_models(cls, item):
+        """Fetch external models based on _get_external_id_relationships.
+
+        This should be called before we call other _validate methods.
+
+        Args:
+            item: ndb.Model. Entity to validate.
+        """
+        multiple_models_keys_to_fetch = {}
+        for field_name_debug, (model_class, keys_to_fetch) in (
+                cls._get_external_id_relationships(item).iteritems()):
+            multiple_models_keys_to_fetch[field_name_debug] = (
+                model_class, keys_to_fetch)
+        fetched_model_instances = (
+            datastore_services.fetch_multiple_entities_by_ids_and_models(
+                multiple_models_keys_to_fetch.values()))
+        for (field_name, (model_class, field_values)), external_models in zip(
+                multiple_models_keys_to_fetch.iteritems(),
+                fetched_model_instances):
+            cls.external_models[field_name] = (
+                model_class, zip(field_values, external_models))
+
+    @classmethod
+    def _get_validation_functions(cls):
+        """Returns the list of validation function to run."""
```

Comment:
Perhaps specify constraints here, e.g. each validation function should accept only a single arg, which is the model instance to validate.